typespec-extension/src/operation-utils.ts

```diff
@@ -17,15 +17,37 @@ import { EmitterOptions } from "./emitter.js";
 import { getNamespace, logWarning, pascalCase } from "./utils.js";
 import { modelIs, unionReferredByType } from "./type-utils.js";
 
-export const specialHeaderNames = new Set([
+export const SPECIAL_HEADER_NAMES = new Set([
   "repeatability-request-id",
   "repeatability-first-sent",
   "x-ms-client-request-id",
   "client-request-id",
   "return-client-request-id",
 ]);
 
-export const originApiVersion = "modelerfour:synthesized/api-version";
+export const ORIGIN_API_VERSION = "modelerfour:synthesized/api-version";
+
+export const CONTENT_TYPE_KEY = "content-type";
+
+export function isKnownContentType(contentTypes: string[]): boolean {
+  // azure-core SerializerEncoding.SUPPORTED_MIME_TYPES
+  const supported_mime_types = new Set<string>([
+    "text/xml",
+    "application/xml",
+    "application/json",
+    "text/css",
+    "text/csv",
+    "text/html",
+    "text/javascript",
+    "text/plain",
+    // not in azure-core
+    "application/merge-patch+json",
+  ]);
+
+  return contentTypes.some((it) => {
+    return supported_mime_types.has(it);
+  });
+}
```

Comment:
I am following java azure-core here. If core does not know how to serialize, we assume it be unknown.